lib/web_ui/lib/src/engine/canvaskit/picture.dart

```diff
@@ -24,14 +30,38 @@ class CkPicture extends ManagedSkiaObject<SkPicture> implements ui.Picture {
   @override
   int get approximateBytesUsed => 0;
 
+  /// Whether the picture has been disposed of.
+  /// 
+  /// This is indended to be used in tests and assertions only.
+  bool get debugIsDisposed => _isDisposed;
+
+  /// This is set to true when [dispose] is called and is never reset back to
+  /// false.
+  /// 
+  /// This extra flag is necessary on top of [rawSkiaObject] because
+  /// [rawSkiaObject] being null does not indicate permanent deletion. See
+  /// similar flag [SkiaObjectBox.isDeletedPermanently].
+  bool _isDisposed = false;
```

Comment:
I appreciate the comment, but how would we ever have a null skia object after initializing this?

Reply:
The parent class (`ManagedSkiaObject`) has a `didDelete` method that nulls it out. It is called when an object is evicted from the cache, at which point the Skia object is deleted. That deletion may not be permanent. The object may still be resurrected. OTOH disposal is permanent.